pkg/util/template/translation.go

```diff
@@ -119,6 +123,37 @@ func (t *translationJSON) viewEffectiveResource(resources []resource.ResourceFil
 	return translationData, nil
 }
 
+func (t *translationJSON) viewAppFile(resources []resource.ResourceFile, view resource.AppFileView) (interface{}, error) {
+	// AppFileView on translation.json returns the translation.json in the app FS if exists.
+	path := view.AppFilePath()
+
+	translationObj := make(map[string]string)
+	for _, resrc := range resources {
+		if resrc.Location.Fs.AppFs() && path == resrc.Location.Path {
+			var jsonObj map[string]interface{}
```

Comment:
Why a round-trip to a `map[string]interface{}`? The ordering/whitespaces would be lost.

We validate the effective file during update, so there should be no need to do it here.